projects/graph/src/graph.py

```diff
@@ -32,6 +59,15 @@ def add_directed_edge(self, v1, v2):
             self.vertices[v1].edges.add(v2)
         else:
             raise IndexError("That vertex does not exist!")
+    def dft(self, starting_node, visited=None):        
+        if visited is None:            
+            visited = []
+        visited.append(starting_node)  
+        print(starting_node)      
+        for node in self.vertices[starting_node].edges:
+            if node not in visited:
+                self.dft(node, visited)
+        return visited
 
```

Comment:
Nice job on the DFT here! The next step would be to use this to graph connect components to each other in order of traversal.